src/osd/PrimaryLogPG.cc

```diff
@@ -6130,11 +6130,12 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	int r = osd->store->fiemap(ch, ghobject_t(soid, ghobject_t::NO_GEN,
 						  info.pgid.shard),
 				   op.extent.offset, op.extent.length, bl);
-	osd_op.outdata = std::move(bl);
+	auto bl_length = bl.length();
+        osd_op.outdata = std::move(bl);
```

Comment:
My only comment is this: if fiemap() returns an error, the 'bl' is left untouched by that call.
But as a created bufferlist has a defined length (of 0), all is well.